web/src/features/time/BottomSheet.ts

```diff
@@ -0,0 +1,340 @@
+      if (newChildrenHeight !== state.childrenHeight) {
+        state.childrenHeight = newChildrenHeight;
+        updateBackdropPosition();
+        updateSnapPoints();
+        adjustCurrentSnapPoint();
+      }
+    }
+  }
+
+  function adjustCurrentSnapPoint(): void {
+    const currentSnapValue = state.snapPointsWithChildHeight[state.currentSnap];
+    if (
+      currentSnapValue === state.childrenHeight ||
+      state.currentSnap === state.snapPointsWithChildHeight.length - 1
+    ) {
+      // If the current snap point is set to the full height of the inner element,
+      // or if it's the last snap point, adjust it to the new height
+      setSnap(state.snapPointsWithChildHeight.length - 1);
+    }
+  }
+
+  function updateSnapPoints(): void {
+    if (props.snapPoints) {
+      const maxSnapPoint = Math.max(...props.snapPoints);
+      state.snapPointsWithChildHeight =
+        state.childrenHeight > maxSnapPoint
+          ? [
+              ...props.snapPoints.filter((point) => point !== maxSnapPoint),
+              state.childrenHeight,
+            ]
+          : [...props.snapPoints];
+    } else {
+      const safeAreaBottom = getSafeAreaBottom();
+      state.snapPointsWithChildHeight = [safeAreaBottom + 60, state.childrenHeight];
+    }
+  }
+
+  function updateBackdropPosition(): void {
+    const currentSheetHeight =
+      Number.parseFloat(element.style.height) || state.snapPointsWithChildHeight[0];
+    const backdropHeight = Math.max(currentSheetHeight - state.childrenHeight + 1, 0);
+    backdropElement.style.height = `${backdropHeight}px`;
+    backdropElement.style.bottom = '0px';
+  }
+
+  function addEventListeners(): () => void {
+    const handlers = createHandlers();
+    const resizeHandler = () => {
+      updateChildrenHeight();
+      setSnap(state.currentSnap);
+    };
+
+    // Touch events
+    element.addEventListener('touchstart', handlers.start, { passive: true });
+    element.addEventListener('touchmove', handlers.move, { passive: false });
+    element.addEventListener('touchend', handlers.end, { passive: true });
+
+    // Mouse events
+    element.addEventListener('mousedown', handlers.start);
+    window.addEventListener('mousemove', handlers.move);
+    window.addEventListener('mouseup', handlers.end);
+
+    window.addEventListener('resize', resizeHandler);
+
+    const resizeObserver = new ResizeObserver(updateChildrenHeight);
+    resizeObserver.observe(childrenElement);
+
+    return () => {
+      element.removeEventListener('touchstart', handlers.start);
+      element.removeEventListener('touchmove', handlers.move);
+      element.removeEventListener('touchend', handlers.end);
+      element.removeEventListener('mousedown', handlers.start);
+      window.removeEventListener('mousemove', handlers.move);
+      window.removeEventListener('mouseup', handlers.end);
+      window.removeEventListener('resize', resizeHandler);
+      resizeObserver.disconnect();
+    };
```

Comment:
Would it be possible to add these directly to the elemetns in a more "react" syle code?

We should be gettng a lot of the logic in here by default then.